src/sage/groups/finitely_presented.py

```diff
@@ -1348,7 +1353,71 @@ def abelian_invariants(self):
         Uses GAP.
         """
         invariants = self.gap().AbelianInvariants()
-        return tuple( i.sage() for i in invariants )
+        return tuple(i.sage() for i in invariants)
+
+    def abelianization(self, ring=QQ):
+        r"""
+        Return the abelianization of ``self`` together with
+        the needed information to recover its group algebra.
+
+        INPUT:
+
+        - ``ring`` -- base ring (default ``QQ``). The base ring for
+          the group algebra of ``self``.
```

Comment:
```suggestion
        - ``ring`` -- (default: ``QQ``); the base ring for
          the group algebra of ``self``
```
Although I don't understand why this is needed. I feel the whole output of the method should just be the abelianization of `self` and any additional construction should be constructed from the output or a different method. As a user using this method, I would be very surprised by the output.